### ISSUE TITLE
Build failed on Ubuntu 20.04.4 LTS

### ISSUE THREAD

**Issue by jafl** [Genivia/RE-flex#131]
After a clean checkout of v3.2.2:

```2022-03-27T19:31:35.3170040Z Cleaning reflex
2022-03-27T19:31:35.3200847Z make[2]: Entering directory '/home/runner/work/reflex/lib'
2022-03-27T19:31:35.3201539Z rm -rf *.dSYM
2022-03-27T19:31:35.3242404Z rm -f *.o *.gch *.log
2022-03-27T19:31:35.3284543Z rm -f *.a *.la *.so *.dylib
2022-03-27T19:31:35.3326306Z make[2]: Leaving directory '/home/runner/work/reflex/lib'
2022-03-27T19:31:35.3329944Z /home/runner/work/reflex
2022-03-27T19:31:35.3356011Z make[2]: Entering directory '/home/runner/work/reflex/src'
2022-03-27T19:31:35.3356707Z rm -rf *.dSYM
2022-03-27T19:31:35.3397332Z rm -f *.o *.gch *.log
2022-03-27T19:31:35.3437884Z rm -f reflex ../bin/reflex
2022-03-27T19:31:35.3458557Z make[2]: Leaving directory '/home/runner/work/reflex/src'
2022-03-27T19:31:35.3461703Z /home/runner/work/reflex
2022-03-27T19:31:35.3487968Z make[2]: Entering directory '/home/runner/work/reflex/unicode'
2022-03-27T19:31:35.3488588Z rm -rf *.dSYM
2022-03-27T19:31:35.3529334Z rm -f *.o *.gch *.log
2022-03-27T19:31:35.3569895Z rm -f reflex
2022-03-27T19:31:35.3591070Z make[2]: Leaving directory '/home/runner/work/reflex/unicode'
2022-03-27T19:31:35.3594335Z /home/runner/work/reflex
2022-03-27T19:31:35.3622992Z make[2]: Entering directory '/home/runner/work/reflex/tests'
2022-03-27T19:31:35.3623480Z rm -rf *.dSYM
2022-03-27T19:31:35.3662102Z rm -f *.o *.gch *.log
2022-03-27T19:31:35.3703069Z rm -f lex.yy.h lex.yy.cpp y.tab.h y.tab.c
2022-03-27T19:31:35.3723547Z rm -rf *.dSYM
2022-03-27T19:31:35.3764237Z rm -f *.o *.gch *.log
2022-03-27T19:31:35.3805244Z rm -f lex.yy.h lex.yy.cpp y.tab.h y.tab.c reflex.*.cpp reflex.*.gv reflex.*.txt
2022-03-27T19:31:35.3846351Z rm -f a.out test_regex_history dump.gv dump.pdf dump.cpp
2022-03-27T19:31:35.3867091Z rm -f lorem streams test rtest ptest btest stest test_bits test_ranges
2022-03-27T19:31:35.3887740Z make[2]: Leaving directory '/home/runner/work/reflex/tests'
2022-03-27T19:31:35.3891281Z /home/runner/work/reflex
2022-03-27T19:31:35.3920940Z make[2]: Entering directory '/home/runner/work/reflex/examples'
2022-03-27T19:31:35.3921560Z rm -rf *.dSYM
2022-03-27T19:31:35.3963057Z rm -f *.o *.gch *.log
2022-03-27T19:31:35.4006579Z rm -f lex.yy.h lex.yy.hpp lex.yy.cpp *.tab.h *.tab.c *.tab.hxx *.tab.cxx parser.hpp parser.cpp scanner.hpp scanner.cpp location.hh location.hpp position.hh position.hpp stack.hh stack.hpp reflex.*.cpp reflex.*.gv reflex.*.txt
2022-03-27T19:31:35.4055857Z rm -f flexexample? reflexexample? flexexample?xx reflexexample?xx
2022-03-27T19:31:35.4099773Z rm -f flexexample?? reflexexample?? flexexample??xx reflexexample??xx
2022-03-27T19:31:35.4148068Z rm -f ctokens jtokens ptokens echo readline calc wc wcu wcpp wcwc tag tag_lazy tag_lazystates tag_unicode tag_tidy cow cows indent1 indent2 json yaml braille unicode csv scanstrings yyscanstrings mmap fastfind fastsearch cards cvt2utf ugrep gz dos url_boost wc_boost url_pcre2 wc_pcre2 minic minicdemo.class
2022-03-27T19:31:35.4170651Z make[2]: Leaving directory '/home/runner/work/reflex/examples'
2022-03-27T19:31:35.4174470Z /home/runner/work/reflex
2022-03-27T19:31:35.4195783Z 
2022-03-27T19:31:35.4196150Z OK
2022-03-27T19:31:35.4225501Z 
2022-03-27T19:31:35.4225867Z Building reflex...
2022-03-27T19:31:35.4226213Z 
2022-03-27T19:31:38.7780238Z Compiling reflex with AVX2 optimizations
2022-03-27T19:31:38.7780768Z 
2022-03-27T19:31:38.7832983Z make[2]: Entering directory '/home/runner/work/reflex/lib'
2022-03-27T19:31:38.7833811Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c convert.cpp
2022-03-27T19:31:38.7835062Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c debug.cpp
2022-03-27T19:31:38.7840641Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c error.cpp
2022-03-27T19:31:38.7857622Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c input.cpp
2022-03-27T19:31:38.7897441Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c ../unicode/block_scripts.cpp
2022-03-27T19:31:38.7920224Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c ../unicode/language_scripts.cpp
2022-03-27T19:31:38.7952077Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c ../unicode/letter_scripts.cpp
2022-03-27T19:31:38.7979087Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c matcher.cpp
2022-03-27T19:31:38.7997423Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c matcher_avx2.cpp
2022-03-27T19:31:38.8007223Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c matcher_avx512bw.cpp
2022-03-27T19:31:38.8069754Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c pattern.cpp
2022-03-27T19:31:38.8127825Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c posix.cpp
2022-03-27T19:31:38.8156737Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c simd_avx2.cpp
2022-03-27T19:31:38.8195096Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c simd_avx512bw.cpp
2022-03-27T19:31:38.8373294Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c unicode.cpp
2022-03-27T19:31:38.8404761Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -c utf8.cpp
2022-03-27T19:32:00.2243745Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -shared -o libreflex.so -fPIC convert.cpp debug.cpp error.cpp input.cpp ../unicode/block_scripts.cpp ../unicode/language_scripts.cpp ../unicode/letter_scripts.cpp matcher.cpp matcher_avx2.o matcher_avx512bw.o pattern.cpp posix.cpp simd_avx2.o simd_avx512bw.o unicode.cpp utf8.cpp
2022-03-27T19:32:00.2270526Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include -mavx2 -DHAVE_AVX2 -shared -o libreflexmin.so -fPIC debug.cpp error.cpp input.cpp matcher.cpp matcher_avx2.o matcher_avx512bw.o pattern.cpp simd_avx2.o simd_avx512bw.o utf8.cpp
2022-03-27T19:32:00.6952476Z ar -rsc libreflex.a convert.o debug.o error.o input.o block_scripts.o language_scripts.o letter_scripts.o matcher.o matcher_avx2.o matcher_avx512bw.o pattern.o posix.o simd_avx2.o simd_avx512bw.o unicode.o utf8.o
2022-03-27T19:32:00.6959407Z ar -rsc libreflexmin.a debug.o error.o input.o matcher.o matcher_avx2.o matcher_avx512bw.o pattern.o simd_avx2.o simd_avx512bw.o utf8.o
2022-03-27T19:32:00.7067289Z ranlib libreflex.a
2022-03-27T19:32:00.7125602Z ranlib libreflexmin.a
2022-03-27T19:32:15.1778956Z /usr/bin/ld: matcher_avx2.o: relocation R_X86_64_PC32 against symbol `_ZN6reflex2HWE' can not be used when making a shared object; recompile with -fPIC
2022-03-27T19:32:15.1779488Z /usr/bin/ld: final link failed: bad value
2022-03-27T19:32:15.1784849Z collect2: error: ld returned 1 exit status
2022-03-27T19:32:15.1908976Z make[2]: *** [Make:44: libreflexmin.so] Error 1
2022-03-27T19:32:15.1909384Z make[2]: *** Waiting for unfinished jobs....
2022-03-27T19:32:20.5405969Z /usr/bin/ld: matcher_avx2.o: relocation R_X86_64_PC32 against symbol `_ZN6reflex2HWE' can not be used when making a shared object; recompile with -fPIC
2022-03-27T19:32:20.5406494Z /usr/bin/ld: final link failed: bad value
2022-03-27T19:32:20.5407782Z collect2: error: ld returned 1 exit status
2022-03-27T19:32:20.5501643Z make[2]: *** [Make:41: libreflex.so] Error 1
2022-03-27T19:32:20.5502392Z make[2]: Leaving directory '/home/runner/work/reflex/lib'
2022-03-27T19:32:20.5560450Z 
2022-03-27T19:32:20.5560839Z Installing reflex...
2022-03-27T19:32:20.5561027Z 
2022-03-27T19:32:20.5577121Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include  -shared -o libreflex.so -fPIC convert.cpp debug.cpp error.cpp input.cpp ../unicode/block_scripts.cpp ../unicode/language_scripts.cpp ../unicode/letter_scripts.cpp matcher.cpp matcher_avx2.o matcher_avx512bw.o pattern.cpp posix.cpp simd_avx2.o simd_avx512bw.o unicode.cpp utf8.cpp
2022-03-27T19:32:20.5579202Z c++ -Wall -Wunused -Wextra -O2 -I. -I../include  -shared -o libreflexmin.so -fPIC debug.cpp error.cpp input.cpp matcher.cpp matcher_avx2.o matcher_avx512bw.o pattern.cpp simd_avx2.o simd_avx512bw.o utf8.cpp
2022-03-27T19:32:26.7847255Z /usr/bin/ld: matcher_avx2.o: relocation R_X86_64_PC32 against undefined symbol `_ZN6reflex2HWE' can not be used when making a shared object; recompile with -fPIC
2022-03-27T19:32:26.7847959Z /usr/bin/ld: final link failed: bad value
2022-03-27T19:32:26.7848366Z collect2: error: ld returned 1 exit status
2022-03-27T19:32:26.7852340Z make: *** [Make:44: libreflexmin.so] Error 1
2022-03-27T19:32:26.7852794Z make: *** Waiting for unfinished jobs....
2022-03-27T19:32:31.6380194Z /usr/bin/ld: matcher_avx2.o: relocation R_X86_64_PC32 against undefined symbol `_ZN6reflex2HWE' can not be used when making a shared object; recompile with -fPIC
2022-03-27T19:32:31.6380725Z /usr/bin/ld: final link failed: bad value
2022-03-27T19:32:31.6382612Z collect2: error: ld returned 1 exit status
2022-03-27T19:32:31.6391362Z make: *** [Make:41: libreflex.so] Error 1
```

**Comment by genivia-inc** [comment]
Thanks for the feedback. The build steps have changed again, causing this issue.

I believe the fix is to change `lib/Make` as follows by renaming the .o files to .cpp files:
~~~
libreflex.so:		convert.cpp debug.cpp error.cpp input.cpp ../unicode/block_scripts.cpp ../unicode/language_scripts.cpp ../unicode/letter_scripts.cpp matcher.cpp matcher_avx2.cpp matcher_avx512bw.cpp pattern.cpp posix.cpp simd_avx2.cpp simd_avx512bw.cpp unicode.cpp utf8.cpp
			$(CPP) $(CFLAGS) -shared -o $@ -fPIC $^

libreflexmin.so:	debug.cpp error.cpp input.cpp matcher.cpp matcher_avx2.cpp matcher_avx512bw.cpp pattern.cpp simd_avx2.cpp simd_avx512bw.cpp utf8.cpp
			$(CPP) $(CFLAGS) -shared -o $@ -fPIC $^
~~~
Can you please verify this also? Run `make -f Make` in `lib` to check. If this works then `./build.sh` should work too, as well as linking the .so files.

**Comment by genivia-inc** [comment]
Should be fixed in the v3.2.3 update available now.

**Comment by jafl** [comment]
Sorry I didn't get a chance to test it for you earlier.  v3.2.3 does fix the issue.  Thanks!
